pkg/asset/machines/openstack/machines.go

```diff
@@ -337,6 +363,13 @@ func failureDomainsFromSpec(mpool openstack.MachinePool) []machinev1.OpenStackFa
 					AvailabilityZone: mpool.RootVolume.Zones[i],
 				}
 			}
+
+			switch len(mpool.RootVolume.Types) {
+			case 0:
+				failureDomains[i].RootVolume.VolumeType = mpool.RootVolume.Type
+			default:
```

Comment:
I think you are right above treating "one zone" and "one volume type" as acceptable cases, whatever the number of counterparts. I believe there should be a `case 1` here, where the same type is used for all failure domains (just like we do above for zones).

Reply:
Good point and it's aligned with "supporting 1 type in the new `Types`". I'll add that.